src/main/java/com/kodilla/ecommercee/service/CartService.java

```diff
@@ -0,0 +1,71 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.exception.ResourceNotExistException;
+import com.kodilla.ecommercee.repository.CartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+public class CartService {
+
+    private final CartRepository cartRepository;
+
+    public BigDecimal calculateValueOfProducts(List<Product> products) {
+        return products.stream()
+                .map(Product::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    public Cart createEmptyCart() {
+        Cart cart = new Cart(new BigDecimal("0"));
+        return cartRepository.save(cart);
+    }
+    public Cart getCart(long cartId) throws ResourceNotExistException {
+        return cartRepository.findById(cartId).orElseThrow(() -> new ResourceNotExistException("Cart not exists"));
+    }
+
+    public List<Product> getProductsFromCart(long cartId) throws ResourceNotExistException {
+        return getCart(cartId).getProducts();
+    }
+
+    public List<Product> addProductToCart(long cartId, Product product) throws ResourceNotExistException {
+        Cart findCart = getCart(cartId);
+
+        List<Product> products = findCart.getProducts();
+        products.add(product);
+
+        BigDecimal cartValue = calculateValueOfProducts(products);
+        findCart.setValue(cartValue);
+        cartRepository.save(findCart);
+
+        return products;
+    }
+
+    public List<Product> removeProductFromCart(long cartId, long productId) throws ResourceNotExistException {
```

Comment:
dlaczego masz tu listę skoro usuwasz tylko jeden produkt?

Reply:
właściwie to mógłby być void (albo nawet powinien)

Reply:
void mi nie potwierdzi, że w koszyku są tylko te produkty, które miały tam być

Reply:
ale potwierdzi Ci to inny endpoint w razie czego. Nie ma potrzeby zwracać pozostałych produktów, bo to generuje niepotrzebny ruch na bazie.